src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -98,15 +98,16 @@ def validate_global_constraints(self):
     @property
     def kernel_references(self):
         '''
-        Returns a set or a 2-tuple of sets of the input (first entry) and
-        output (second entry) references (whether to arrays or objects)
-        required by the Kernel call(s) that are children of this directive.
-        This is the set of quantities that must be available on the remote
+        Returns a 1-tuple or a 2-tuple of sets depending on the working API.
+        If a 1-tuple, the set includes both input and output references
+        (whether to arrays or objects) required by the Kernel call(s) that are
+        children of this directive. If a 2-tuple, the first entry is the set of
+        input references and the second entry is the set of output references.
+        The set(s) describe the quantities that must be available on the remote
         device (probably a GPU) before the parallel region can be begun.
 
-        :returns: set or 2-tuple of input and output sets of variable names
-        :rtype: Set[str] or Tuple[Set[:py:class:`psyclone.core.Signature`],
-                                  Set[:py:class:`psyclone.core.Signature`]]
+        :returns: 1-tuple or 2-tuple of input and output sets of variable names
+        :rtype: Tuple[Set[:py:class:`psyclone.core.Signature`]]
```

Comment:
`...], Optional[Set[:py:class:`...`]]]`

Reply:
I thought a bit about it, what's the rule? Of course, if this were a 20-tuple or an unknown-dim-tuple of `str` say, you would rather type `Tuple[str]`.  In this case, it is either a one- or a two-tuple, but it feels odd to say it's an option...

Reply:
Done, but used `Union` instead to the same effect.
And, for completeness, I was wrong. An tuple of strings of arbitrary length is denoted `Tuple[str, ...]`.
I was confused because, for instance, lists or sets of strings of arbitrary length are typed `List[str]` or `Set[str]` instead. I can't say I fully understand that PEP, I find it hard to read with many examples of usage before definition.